packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleTextViewer/SingleTextViewerContainer.js

```diff
@@ -1,38 +1,52 @@
 import PropTypes from 'prop-types'
 import asyncStates from '@zooniverse/async-states'
 
+import { useTextData } from '@hooks'
+import locationValidator from '../../helpers/locationValidator'
 import SingleTextViewer from './SingleTextViewer'
 
+const defaultSubject = {
+  id: '',
+  locations: []
+}
+
 export default function SingleTextViewerContainer ({
-  content = '',
-  contentLoadingState = asyncStates.initialized,
-  error = null,
   height = '',
+  loadingState = asyncStates.initialized,
   onError = () => true,
-  onReady = () => true
+  onReady = () => true,
+  subject = defaultSubject
 }) {
-  if (contentLoadingState === asyncStates.success) {
-    onReady()
+  const textData = useTextData(
+    subject,
+    onReady,
+    onError
+  )
```

Comment:
Not a requirement for this PR, but with the other data-fetching hooks, I've used an API that looks like this:
```js
const { data, error, loading, viewer } = useSubjectJSON({ subject, onReady, onError })
const { img, error, loading, subjectImage } = useSubjectImage({ src, onReady, onError })
```
The hook accepts an object, rather than a list of arguments, and returns an object containing the loading state (true/false), a ref for the subject viewer, and any error alongside the data.

Is it worth using the same interface here, for consistency? Have a look at those hooks and see what you think.

Reply:
Yes I think this hook should definitely be consistent with the others noted, I'll refactor accordingly. Thanks, great suggestions!

Reply:
The viewer ref probably isn't needed for this hook. I can't imagine why we'd want to record width and height in pixels for a text file.